Tests/FluentAssertions.Equivalency.Specs/DictionarySpecs.cs

```diff
@@ -526,6 +540,23 @@ public void
         trace.Should().Contain("Recursing into dictionary item greeting at object1");
     }
 
+    [Fact]
+    public void
+        When_a_non_generic_dictionary_is_decided_to_be_equivalent_to_expected_without_nesting_trace_is_still_written()
```

Comment:
❓ I don't quite understand what this test is trying to validate

Reply:
ok so all the three tests you talk about is to test the IsRecursive option which was frequently missing coverage and the way you do that is with ExcludeNestedObject

However, it seems my current tests aren't selective enough about it so I will try to improve them.

Reply:
So as I recheck my tests, I am now wondering: what cases would ExcludeNestedObjects succeed while omitting it fails?

Because I tried this:
```cs
[Fact]
    public void When_a_generic_dictionary_is_typed_as_object_without_nesting_it_should_respect_the_runtime_typed()
    {
        // Arrange
        object subject = new Dictionary<string, SomeNestedObject>
        {
            ["greeting"] = new()
            {
                MyString = "identical string",
                MyChild = new SomeNestedObject { MyString = "identical string" }
            }
        };
        object expectation = new Dictionary<string, SomeNestedObject>
        {
            ["greeting"] = new()
            {
                MyString = "identical string",
                MyChild = new SomeNestedObject { MyString = "DIFFERENT STRING" }
            }
        };

        // Act & Assert
        subject.Should().BeEquivalentTo(expectation, options => options.ExcludingNestedObjects());
    }
``` But it failed.

Even more strange is `When_nested_objects_are_excluded_from_collections_it_should_use_simple_equality_semantics` from CollectionSpecs still succeeds if you remove the ExcludingNestedObjects.

So I feel like there's some problems with this option that makes it too weird to test. Should I remove those test and only keep the other part of the pr?

Reply:
Yeah, let's do that 👍